### PR TITLE
Set loggedIn bool to display profile/login nav links correctly.

### PR DESCRIPTION
**Pull request by Rloeppky**
Bool is set to true, if a username is returned from the current user api Kenny made.

### CODE REVIEW

**Comment by hwbit** [comment]
Not sure if this is working. I am not logged in or anything and I go straight to a blank profile page.

image

**Comment by lik5** [comment]
> Not sure if this is working. I am not logged in or anything and I go straight to a blank profile page.
> 
> image

Might be because the token persists in your local storage.... We should look into that and make sure that we're saving the token the right way

File: preppal-fe/src/components/nav-bar/nav-bar.tsx
Comment (lik5) [inline preppal-fe/src/components/nav-bar/nav-bar.tsx:19]: bad boolean, this should be if (token)

File: preppal-fe/src/components/nav-bar/nav-bar.tsx
Comment (lik5) [inline preppal-fe/src/components/nav-bar/nav-bar.tsx:42]: We probably also want to include new recipe and collections as well since they both require a user

File: preppal-fe/src/components/nav-bar/nav-bar.tsx
Comment (Rloeppky) [reply preppal-fe/src/components/nav-bar/nav-bar.tsx]: Just to make the code easier to read I wrote if(token !== null) which would mean the same check.

File: preppal-fe/src/components/nav-bar/nav-bar.tsx
Comment (Rloeppky) [reply preppal-fe/src/components/nav-bar/nav-bar.tsx:40]: These have been added. They didn't want to be in one loggedIn check, so I made three.

**Comment by Rloeppky** [comment]
Two things of note, to test if this is working, you will need to clear some part of your browsers history to erase the existing token.
Second, the navbar does not refresh after a user hits submit on login and so while it would load correctly on next refresh, but it fails to update after login. This will need to be a new ticket.

File: preppal-fe/src/components/nav-bar/nav-bar.tsx
Comment (lik5) [reply preppal-fe/src/components/nav-bar/nav-bar.tsx]: I believe the check for undefined and null are separate. So in this case when you check if it's not null and token is undefined, the boolean value will be true. There is a way to more loosely check so that null and undefined are both covered but just checking if the token is a truthy value is better

Comment (Rloeppky) [reply preppal-fe/src/components/nav-bar/nav-bar.tsx]: Thank you for telling me this, I didn't even know that truthy was something that existed in this language until now. It has been updated accordingly.

**Review by lik5 — APPROVED** [review]
Looks good to me. Let's remove the remember me checkbox though since it has no functionality

**Comment by Rloeppky** [comment]
Merge conflicts have been resolved with new commits, needing new review and approval.

**Review by lik5 — DISMISSED** [review]

**Comment by lik5** [comment]
We want to remove the checkbox for remember me! I will approve right after

**Comment by Rloeppky** [comment]
> @Rloeppky We want to remove the checkbox for remember me! I will approve right after

@lik5, I had removed it, but I guess the merge conflicts from dev yesterday reset it, sorry. It has been removed again and frontend tests cleaned.

NOTE: Dev had been set back yesterday, and it looks like the out of base with dev that git told me to use has recreated all of Ethan's code in my pull request. I think I may need to create a new pull request when I am at school and close this one instead.

**Don't merge, we can talk in class.**

**Comment by Rloeppky** [comment]
Files and commits have become difficult to work with. Closing branch and restarting with cleaned branch.

**Pull request closed without merging**
